### PR TITLE
API Consolidate "Delete from draft" and "Archive" actions

### PR DESCRIPTION
**Pull request by tractorcow**
Merge with https://github.com/silverstripe/silverstripe-framework/pull/6195

Update to support new versioned dataobject behaviour

### CODE REVIEW

**Comment by tractorcow** [comment]
I've updated this based on feedback from @clarkepaul 

Published page:

image

Unpublished page:

image

**Comment by clarkepaul** [comment]
Looks good to me :)

File: javascript/lang/hr.js [hunk 1]
```diff
@@ -1,53 +0,0 @@
-// This file was generated by silverstripe/cow from javascript/lang/src/hr.js.
```
Comment (chillu) [inline javascript/lang/hr.js]: @tractorcow Why did you delete those?

Comment (flamerohr) [inline javascript/lang/hr.js]: They were hangover changes from 3.x merged up to master.

**Comment by tractorcow** [comment]
Tests are passing, but there's still an occasionally failing behat test related to the "Page status" dropdown not being selectable under some circumstances.
